Common/SemanticAnalyzer/src/AstStatementParser.cpp

```diff
@@ -0,0 +1,185 @@
+#include "AstStatementParser.h"
+
+
+std::vector<std::pair<std::string, std::any>>& SEMANTICANALYZER::AstStatementParser::parse()
+{
+    for(auto it = astTree.begin(); it != astTree.end(); ++it)
+    {
+        std::pair<std::string, std::any> statement;
+
+        if(it->value == OBJECT_DECL)
+        {
+            statement.first = OBJECT_DECL;
+            serializeObjectDec(it, std::any_cast<ObjectDecl&>(statement.second));
+        }
+        else if(it->value == RELATION)
+        {
+            statement.first = RELATION;
+            serializeRelation(it, std::any_cast<Relation&>(statement.second));
+        }
+        else if(it->value == NOTE)
+        {
+            statement.first = NOTE;
+            serializeNote(it, std::any_cast<Note&>(statement.second));
+        }
+        else if(it->value == GRAPH)
+        {
+            statement.first = GRAPH;
+            serializeGraph(it, std::any_cast<Graph&>(statement.second));
+        }
+        else if(it->value == DOT_CLOUD)
+        {
+            statement.first = DOT_CLOUD;
+            serializeDotCloud(it, std::any_cast<DotCloud&>(statement.second));
+        }
+
+        programTree.push_back(statement);
+    }
+
+    return programTree;
+}
+
+
+void SEMANTICANALYZER::AstStatementParser::serializeObjectDec(AST::ASTTree::DSFIterator& objectDecIter, ObjectDecl& objectDecl)
+{
+    for(auto it = objectDecIter; it != AST::ASTTree::DSFIterator(); ++it)
+    {
+        if(it->value == SHAPE)
+        {
+            ++it;
+            objectDecl.shape = it->value;
+        }
+        else if(it->value == ID)
+        {
+            ++it;
+            objectDecl.id = it->value;
+        }
+        else if(it->value == PROPERTY)
+        {
+            Property property;
+            serializeProperty(it, property);
+            objectDecl.properties.push_back(property);
```

Comment:
[minor] В проперти много строк, поэтому предлагаю мувать объект, чтобы избежать ненужных копирований
```suggestion
            objectDecl.properties.push_back(std::move(property));
```
Ниже аналогично
И выше строчку ```programTree.push_back(statement);```

Reply:
спасибо, поправил